src/server.js

```diff
@@ -4,7 +4,6 @@ const mongoInit = require("./database");
 dotenv.config({ path: "config.env" });
 mongoInit();
```

Comment:
This isn't production-ready. If for some reason mongo fails to load up or establish a connection, you're in trouble.
A better way to do this would be

```
connectToMongo() 
  .then(() => {
     // mongo connection successful
     // start server here
      app.listen(process.env.PORT, () => {
         console.log(process.env.accessTokenValidity);
         console.log(`Server listening on port ${process.env.PORT}`);
      });

  })
  .catch(e => {
    // mongo failed to load up
   process.exit(1);
   });


```

Reply:
check index.js of database .all mongoose functions are exported. cannot attach then function.

Reply:
So you need to make a function which does the connection and return a Promise.

```
function connectToMongo () {
return new Promise((res, rej) => {
  // connection here 
//resolve if success, else reject
});
```